v0/destinations/delighted/util.js

```diff
@@ -0,0 +1,80 @@
+const axios = require("axios");
+const { CustomError, getValueFromMessage } = require("../../util");
+const { ENDPOINT } = require("./config");
+
+const isValidEmail = email => {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(email).toLowerCase());
+};
+const isValidPhone = phone => {
+  const phoneformat = /^\+[1-9]\d{10,14}$/;
+  return phoneformat.test(String(phone));
+};
+
+const isValidUserId = (channel, userId) => {
+  if (channel === "email") {
+    if (!isValidEmail(userId)) {
+      throw new CustomError("Email format is not correct.", 400);
+    }
+  } else if (channel === "phone") {
+    if (!isValidPhone(userId)) {
+      throw new CustomError("Phone number format must be E.164.", 400);
+    }
+  } else {
+    throw new CustomError("User Id is not matching the channel type.", 400);
+  }
+};
+
+const userValidity = async (channel, Config, userId) => {
+  const payload = {};
+  if (channel === "email") {
+    payload.email = userId;
+  } else if (channel === "phone") {
+    payload.phone = userId;
+  } else {
+    throw new CustomError("Unable to generate payload for GET request.", 400);
```

Comment:
before calling this function, we are already checking for the channel validity. So, we don't need this else check actually. we can refactor this

Reply:
yeah removed this else